### ISSUE TITLE
Infix types

### ISSUE THREAD

**Issue by ethul** [purescript/purescript#416]
Is there a way to declare infix types as follows?

```purescript
data a :+: b = Inl a | Inr b

type f ~> g = forall a. f a -> g a
```

Thanks!

**Comment by AppShipIt** [comment]
Partially resolved by #1890 (except `type`)?

**Comment by garyb** [comment]
Not really, this is just about types, not constructors. I was going to do this too after making it work for data constructors but figured it wasn't worth holding up 0.8.2 for.
